### DISCUSSION TITLE
error executing the script: Required array length too large

### DISCUSSION THREAD

**Discussion by OmarMohamedAero** [freeplane/gradle-plugin#9] [Q&A]
Description of the Issue
This Groovy script aims to zip a MindMap file along with its associated files into a single ZIP archive. However, it encounters an error during execution: **### "Required array length too large."** This issue arises when attempting to create an array that exceeds the allowed maximum size, potentially due to excessive data being processed or an inefficient algorithm.

Code Overview
The script contains several key functions:

getZipBytes: Creates a ZIP byte array from the specified images and the main MindMap.
addZipEntry: Adds each file or directory to the ZIP output stream.
getPathInZip: Determines the path for each file in the ZIP based on existing references and directories.
getBytes: Converts the MindMap to a byte array in XML format.

I need to solve this problem, any file that exceeds 2G will show this error  

addon link : https://github.com/macmarrum/freeplane-PackAndGoGo.git

**Comment by OmarMohamedAero** [comment]
@macmarrum @dpolivaev @HaaiHenkie @dpolivaevUnite Can you help me with that? I tried my best to solve this by updating the script and creating a new add-on, but it did not work.

If you could make a tutorial on how to create an add-on, that would be wonderful!

**Comment by dpolivaev** [comment]
> any file that exceeds 2G will show this error

Let me google it for you

https://www.google.com/search?q=java+maximum+array+size

https://stackoverflow.com/questions/3038392/do-java-arrays-have-a-maximum-size
